src/juju/reducers.js

```diff
@@ -0,0 +1,24 @@
+/* eslint-disable default-case */
+// immer handles the default case so one isn't needed.
+import produce from "immer";
+
+import { actionsList } from "./actions";
+
+export default function jujuReducers(state = {}, action) {
+  return produce(state, draftState => {
+    switch (action.type) {
+      case actionsList.updateModelList:
+        const modelList = action.payload.userModels.map(model => {
+          return {
+            lastConnection: model.lastConnection,
+            name: model.model.name,
+            ownerTag: model.model.ownerTag,
+            type: model.model.type,
+            uuid: model.model.uuid
+          };
+        });
+        draftState.models = { items: modelList };
```

Comment:
Might be worth providing a default for draftState.models.items (you can do this by passing an object to the second param of `produce`, e.g.

```
produce(
  (draft, action) => {
    switch (action.type) {
     ...
    },
  },
  {
    items: []
  }
);
```

then you could just say `draftState.models.items = modelList` here.

Reply:
This didn't work in practice, we can revisit this in a follow-up as it would be a nice feature to have